src/lib/paw-to-openapi-converter/components/auth-converter.ts

```diff
@@ -1,110 +1,173 @@
-            && (example as OpenAPI.ExampleObject).value?.username === this.request.httpBasicAuth.username
-            && (example as OpenAPI.ExampleObject).value?.password
-            && (example as OpenAPI.ExampleObject).value?.password === this.request.httpBasicAuth.password
+            && this.request.httpBasicAuth
+            && summary
+            && summary === BasicCredentialsLabel
+            && value?.username
+            && value?.username === this.request.httpBasicAuth.username
+            && value?.password
+            && value?.password === this.request.httpBasicAuth.password
           ) {
-            found = true
-            this.key = key.toString()
+            found = true;
+            this.key = key.toString();
           }
-        })
+        });
       }
 
       if (!found) {
-        this.key = `BasicAuth${this.existingExamples.length > 0 ? `_${this.existingExamples.length}` : ''}`
-
-        const securityScheme: OpenAPI.SecuritySchemeObject = {
-          type: "http",
-          scheme: "basic"
-        }
+        this.key = `BasicAuth${this.existingExamples.length > 0 ? `_${this.existingExamples.length}` : ''}`;
 
         const securityExample: OpenAPI.ExampleObject = {
           summary: BasicCredentialsLabel,
-          value: this.request.httpBasicAuth
-        }
-
-        this.scheme = securityScheme
-        this.example = securityExample
+          value: this.request.httpBasicAuth,
+        };
+
+        this.scheme = {
+          type: 'http',
+          scheme: 'basic',
+        };
+        this.example = securityExample;
       }
 
-      this.requirement[this.key] = []
+      this.requirement[this.key] = [];
 
       this.authFound = true;
     }
   }
 
-  private parseHttpBearerAuth() {
-    /**
-     * @TODO
-     */
-  }
+  // eslint-disable-next-line class-methods-use-this
+  private parseHttpBearerAuth(): void {
+    const authHeader = this.request.getHeaderByName('authorization');
+
+    if (authHeader && (authHeader as string).match(/bearer /i)) {
+      this.requirement = {};
+      this.key = 'BearerAuth';
+
+      this.scheme = {
+        type: 'http',
+        scheme: 'bearer',
+      };
+
+      this.requirement[this.key] = [];
 
-  private parseApiKeyAuth(parametersConverters: ParametersConverter) {
-    /**
-     * @TODO
-     * - in query
-     * - in header
-     * - in cookie
-     */
+      this.authFound = true;
+    }
   }
 
-  private parseOAuth2Auth() {
-    /**
-     * @TODO
-     */
+  // eslint-disable-next-line class-methods-use-this
+  private parseOAuth2Auth(): void {
+    const { oauth2 } = this.request;
+
+    if (oauth2) {
+      const {
+        scope,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        authorization_uri,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        access_token_uri,
+        ...restOauth2Values
+      } = oauth2;
+
+      this.requirement = {};
+      this.key = 'OAuth2';
```

Comment:
This is quite incorrect. If one has two _**very different**_ authentication methods using OAuth 2 (or Basic Auth for that matter), they are mixed up in the output `securitySchemes`.
This is an important use case. At the same time, we don't have to create X entries in `securitySchemes`, if they are all the same.

A suggestion would be to:
1. Create a simple hash of the security scheme (for example, can be first few characters of the md5 of `authorization_uri + authorization_uri + scope`)
2. Use this hash as key in addition to OAuth 2, you could name the key `OAuth2_{hash}`
3. A bonus would be: right before finishing the import process, to check how many auths of one kind do we have. If only one `OAuth2`, then rename it from `OAuth2_{hash}` to simply `OAuth2` to keep a nice output

Open to any other implementation suggestions, as long as we keep all the possible `securitySchemes` if they end up being different.

Reply:
@mittsh I would just want to discuss one thing here - OpenAPI documentation page have this authorization form view looking like here:
<img width="662" alt="image" src="https://user-images.githubusercontent.com/69789351/91987864-e08f7d00-ed2e-11ea-91a9-3efb35f2ae08.png">

That "key" value is used here as form label, so using "OAuth2 {hash}" might look strange for user. Maybe we could use something more human-readable like "OAuth2 {request_name}"? What do You think about it @mittsh ?

Reply:
@wch-mirek That's a good point! Though, I'm afraid that if a user has the exact same auth mechanism in 20 different requests, we'll create 20 different entries for OAuth 2 when we could have a single one. That's where the idea of the hash came from. Using a hash makes sure that:
1. Two different auths will never be mixed
2. If many requests use the same auth, we don't recreate auth configs needlessly

Perhaps you could use a hash internally to have this behavior but just use `{request_name}` as a label for the user (picking the 1st request that used this auth).

This solution would work for me if you like it as well 😄

Reply:
@mittsh Hm ok I think that tool is mostly for developers, so hashes in labels shouldn't be that scary and also I tried to come up with solution of putting 1st request name as authorization label and instead of hashing values - simple compare values. But I stumble on big if like so:

```
if (
  !sameAuthFound
  && oauth2?.grant_type === AuthConverter.capitalToCamelCase(grantType)
  && oauth2?.authorization_uri === (flow as OpenAPI.OAuthFlowObject).authorizationUrl
  && oauth2?.access_token_uri === (flow as OpenAPI.OAuthFlowObject).tokenUrl
  && oauth2.scope === AuthConverter.convertOpenAPIToPawScopes(
    (flow as OpenAPI.OAuthFlowObject).scopes,
  )
)
```

Using hashes will be more elegant and it will shorten logic to comparing just labels for authorizations. Then I'll go here with 'OAuth2 {hash}' solution then :)

Reply:
Good 👍